### PR TITLE
Add custom fields and custom field options streams

### PR DESCRIPTION
**Pull request by rliddler**
This PR adds the custom fields stream and in an effort to normalize slightly tables together it queries for each one the custom field options that are available and serializes them into that schema.

### CODE REVIEW

File: model/custom_field_v2.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+var CustomFieldV2 customFieldV2
+
+func (customFieldV2) Schema() Property {
+	return Property{
+		Types: []string{"object"},
+		Properties: map[string]Property{
+			"id": {
+				Types: []string{"string"},
+			},
+			"catalog_type_id": {
```
Comment (isaacseymour) [inline model/custom_field_v2.go:16]: This suggests that we should put catalog _types_ into the warehouse, at least (or remove this property)

File: model/custom_field_v2.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			"created_at": DateTime.Schema(),
+			"updated_at": DateTime.Schema(),
+			"options":    Optional(ArrayOf(CustomFieldOptionV1.Schema())),
+		},
+	}
+
+}
+
+func (customFieldV2) Serialize(input client.CustomFieldV2, customFieldOptions []client.CustomFieldOptionV1) map[string]any {
+	var options []map[string]any
```
Comment (isaacseymour) [inline model/custom_field_v2.go:37]: I would personally go for not storing these on the `custom_fields` table, but putting them on their own table

File: tap/stream_custom_fields.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	)
+
+	for {
+		page, err := cl.CustomFieldOptionsV1ListWithResponse(ctx, &client.CustomFieldOptionsV1ListParams{
+			CustomFieldId: id,
+			PageSize:      &pageSize,
+			After:         after,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "listing incidents")
```
Comment (isaacseymour) [inline tap/stream_custom_fields.go:76]: not incidents

**Review by isaacseymour — APPROVED** [review]

File: model/custom_field_v2.go (hunk 2 above)
Comment (rliddler) [reply model/custom_field_v2.go:37]: Yeah I've gone back and forth on this one - I kind of agree, as this data is also normalized into the incidents table (in this kind of format) but this is more a stream of the original data.

Will split it into 2 streams.

File: model/custom_field_v2.go (hunk 1 above)
Comment (rliddler) [reply model/custom_field_v2.go:16]: I've put a big "maybe" on the catalog API and was going to leave it off for now - agree would be good to hide these for now.

**Pull request merged**